pkg/apis/k0s.k0sproject.io/v1beta1/calico.go

```diff
@@ -23,6 +23,9 @@ type Calico struct {
 	// Enable wireguard-based encryption (default: false)
 	EnableWireguard bool `json:"wireguard"`
 
+	// Environment variables to configure Calico node (see https://docs.projectcalico.org/reference/node/configuration)
+	EnvVars map[string]string `json:"envVars,omitempty"`
```

Comment:
Would be good to have some validation for these à la [`ValidateEnv`](https://pkg.go.dev/k8s.io/kubernetes@v1.25.3/pkg/apis/core/validation#ValidateEnv). Maybe there's a way to funnel these values through that function.

Reply:
TBH, I don't see how we could validate here anything. Env var's name and value are just strings, and only strings are allowed here.

Reply:
Env vars cannot have arbitrary names. The function I mentioned basically calls this code snippet:

https://github.com/kubernetes/kubernetes/blob/1ffc17e329f9ba3dbe840a7f37513eb8ef074740/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L403-L417

I though about something along those lines:

```go
import (
	"k8s.io/apimachinery/pkg/util/validation"
	"k8s.io/apimachinery/pkg/util/validation/field"
)

var invalidNames []string
for name := range c.EnvVars {
	if msgs := validation.IsEnvVarName(name); len(msgs) > 0 {
		invalidNames = append(invalidNames, name)
	}
}
if invalidNames != nil {
	return field.Invalid(field.NewPath("envVars"), invalidNames, "invalid environment variable names")
}
```